content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx

```diff
@@ -0,0 +1,193 @@
+---
+title: 'Comparing columns of same table through raw queries'
+metaTitle: 'Comparing columns of same table through raw queries'
+metaDescription: 'Comparing columns of same table through raw queries'
+---
+
+## Workaround
+
+Comparing columns from a same table is a common scenario which developers have to deal with. Some examples of it could be comparing two numeric values in a same table or comparing two dates in a same table.
+
+
+## Solution
+
+This could be achieved by using [Raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)
+
+A use case could be of getting posts which has more comments than likes, In this case we need to compare commentsCount and likesCount.
+
+```prisma
+model Post {
+  id            Int      @id @default(autoincrement())
+  createdAt     DateTime @default(now())
+  updatedAt     DateTime @updatedAt
+  title         String   
+  content       String?
+  published     Boolean  @default(false)
+  author        User     @relation(fields: [authorId], references: [id])
+  authorId      Int
+  likesCount    Int
+  commentsCount Int
+}
+```
+
+Queries depending upon databases could look something like:
+
+*PostgreSQL / CockroachDB*
+
+```js
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function initiateNumbersComparisonRawQuery() {
+  const response =
+    await prisma.$queryRaw`SELECT * FROM "public"."Post" WHERE "likesCount" < "commentsCount";`;
+
+  console.log(response);
+}
+
+await initiateNumbersComparisonRawQuery();
+```
+
+*MySQL*
+
+```js
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function initiateNumbersComparisonRawQuery() {
+  const response =
+    await prisma.$queryRaw`SELECT * FROM "public"."Post" WHERE "likesCount" < "commentsCount";`;
+
+  console.log(response);
+}
+
+await initiateNumbersComparisonRawQuery();
+```
+
+*Sqlite*
+
+```js
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function initiateNumbersComparisonRawQuery() {
+  const response =
+    await prisma.$queryRaw`SELECT * FROM "Post" WHERE "likesCount" < "commentsCount";`;
+
+  console.log(response);
+}
+
+await initiateNumbersComparisonRawQuery();
+```
+
+Executing the above queries depending upon the database would filter posts which has less likes compared to comments.
```

Comment:
There's a guideline in tech writing to avoid using the word "execute" unless absolutely needed. In. this case, we can say "Running the above queries...." 

Also, in this sentence we want "that" not "which" (because the clause is important to the entire meaning).

And, we need to put in parentheses the clause "depending upon the database".... and finally, use "fewer" instead of "less" (weird grammar rule about if you can count them, use "fewer").

So final edited sentence would be:
"Running the above queries (depending upon the database) would filter posts that has fewer likes compared to comments.

Reply:
Thanks Tana for the amazing suggestions.
I have updated the sentences.
I think we are good to go. All comments are resolved.